src/components/PetsList.jsx

```diff
@@ -1,19 +1,87 @@
 import FilterList from "./FilterList";
 import PetCard from "./PetCard";
 import { PetItems } from "../files/PetsData";
-import React from "react";
+import React, { useState } from "react";
+import CategoriesFilterCollapse from "./CategoriesFilterCollapse";
 
 const PetsList = () => {
+  const [filteredItems, setFilteredItems] = useState([]);
+  const [isFiltering, setIsFiltering] = useState(false);
+  const [filterItems, setFilterItems] = useState([
+    { type: "species", name: "Cat", checked: true },
+    { type: "species", name: "Dog", checked: true },
+    { type: "species", name: "Other", checked: true }
+  ]);
+  //TODO: These codes are only for april demo purpose only, once we have our service these code will be removed.
+  const filterServiceList = itemUpdated => {
+    let finalItems = [];
+    const checkedspeciesType = itemUpdated.filter(
+      item => item.checked === true && item.type === "species"
+    );
+    setIsFiltering(true);
+    let items = [...PetItems];
+    let attributes = items.filter(i => i.attributes);
+    for (var i in attributes) {
+      var attributesItems = attributes[i].attributes;
+      for (var j in checkedspeciesType) {
+        let type = checkedspeciesType[j].type.toLocaleLowerCase();
+        let name = checkedspeciesType[j].name.toLocaleLowerCase();
+        if (
+          attributesItems.find(
+            x =>
+              x.label.toLocaleLowerCase() === `${type}` &&
+              x.value.toLocaleLowerCase() === `${name}`
+          )
+        ) {
+          finalItems.push(attributes[i]);
+        }
+      }
+      setFilteredItems(finalItems);
+    }
+  };
+
+  const handlePetFilterChange = changeEvent => {
+    setIsFiltering(false);
+    const { checked, name } = changeEvent.target;
+    const itemUpdated = filterItems.map(item => {
+      if (item.name.toLocaleLowerCase() === name.toLocaleLowerCase())
+        return { ...item, checked: checked };
+      return item;
+    });
+    setFilterItems(itemUpdated);
+    const checkedCount = itemUpdated.filter(item => item.checked).length;
+    const isTrue =
```

Comment:
what is true?

Reply:
isTrue contains value either  True or False

Reply:
if you read this in code it sounds like this is always true

```js
if (isTrue) {
 // this will always happen
...
```

where as something like this make it more clear what's happening.

```js
if (isItemUpdated) {
  // suggests that we want to do something when the item is updated
...
```

or 

```js
if (hasActiveFilters) {
  // suggests we have active filters and there is something we want to do wtih them.
...
```

does that make sesne?

Reply:
done